test/k8sT/Istio.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sTest
+
+import (
+	"fmt"
+
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+var _ = Describe("K8sValidatedIstioTest", func() {
+
+	var (
+		kubectl          *helpers.Kubectl
+		logger           *logrus.Entry
+		microscopeErr    error
+		microscopeCancel func() error
+		ciliumPodK8s1    string
+		ciliumFilter     string = "-l k8s-app=cilium"
+		ciliumPath       string = helpers.ManifestGet("cilium_ds_istio.yaml")
+		istioManifest    string = helpers.ManifestGet("istio-auth.yaml")
+		istioNS          string = "istio-system"
+		istioFilter      string = "-l zgroup=istio"
+		istioVersion     string = "release-0.8-20180521-15-16"
+		istioURL         string = fmt.Sprintf("https://storage.googleapis.com/istio-prerelease/daily-build/%[1]s/istio-%[1]s-linux.tar.gz", istioVersion)
+		istioConfigMap   string = helpers.ManifestGet("istio-kube-inject-template.yaml")
+		istioInitPolicy  string = helpers.ManifestGet("istio-sidecar-init-policy.yaml")
+		demoPath         string = helpers.ManifestGet("demo_big.yaml")
+		container        string = "app-frontend"
+	)
+
+	BeforeAll(func() {
+		logger = log.WithFields(logrus.Fields{"testName": "K8sValidatedIstioTest"})
+		logger.Info("Starting")
+
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+		kubectl.Apply(ciliumPath)
+
+		_ = kubectl.Exec(fmt.Sprintf(
```

Comment:
Why does Cilium need to be deleted here? Add comments to indicate why this is occurring.